packages/runtimes/service-worker/src/ServiceWorkerRuntime.js

```diff
@@ -0,0 +1,50 @@
+// @flow
+import {Runtime} from '@parcel/plugin';
+import {urlJoin} from '@parcel/utils';
+
+export default (new Runtime({
+  apply({bundle, bundleGraph}) {
+    if (bundle.env.context !== 'service-worker') {
+      return [];
+    }
+
+    let hasDep;
+    bundle.traverse((node, _, actions) => {
+      if (
+        node.type === 'dependency' &&
+        node.value.specifier === '@parcel/service-worker'
+      ) {
+        hasDep = true;
+        actions.stop();
+      }
+    });
+
+    if (!hasDep) {
+      return [];
+    }
+
+    let manifest = [];
+    bundleGraph.traverseBundles(b => {
+      if (b.bundleBehavior === 'inline' || b.id === bundle.id) {
+        return;
+      }
+
+      manifest.push(urlJoin(b.target.publicUrl, b.name));
+    });
+
+    let code = `import {_register} from '@parcel/service-worker';
```

Comment:
So this imports `@parcel/service-worker` relative to the runtime? It should rather be resolved from the user's package. Yarn 2 would still break